bigqueue.go

```diff
@@ -106,16 +106,23 @@ func (q *MmapQueue) Close() error {
 	return retErr
 }
 
-func (q *MmapQueue) flushIfRequired() error {
-	clock := q.conf.clock
-	shouldFlush := q.mutateOpsSinceFlush >= q.conf.flushIntervalMutateOps
-	shouldFlush = shouldFlush || clock.Since(q.prevFlushTime) >= q.conf.flushElapsedDuration
-	if shouldFlush {
-		if err := q.am.flush(); err != nil {
-			return err
-		}
-		q.mutateOpsSinceFlush = 0
-		q.prevFlushTime = clock.Now()
+// Flush syncs the in memory content of bigqueue to disk
+func (q *MmapQueue) Flush() error {
```

Comment:
should we add a call to Flush as part of this test [TestResetOfBothFlushIntervalStates](https://github.com/grandecola/bigqueue/blob/master/bigqueue_flush_test.go#L151)?
just to assert that either of: enoughTime, enoughMutate, or explicit flush results in a reset of the flush interval state.